src/v/cluster_link/link.h

```diff
@@ -45,5 +46,6 @@ class link {
     chunked_hash_map<ss::sstring, std::unique_ptr<task>> _tasks;
     model::metadata _config;
     kafka::data::rpc::partition_leader_cache* _partition_leader_cache;
+    kafka::data::rpc::partition_manager* _partition_manager;
```

Comment:
nit: the same comment as for the `leaders_cache` if this is going to be a generic building block let's maybe move it out from the `rpc` namespace